src/bsoncxx/types/bson_value/value.hpp

```diff
@@ -38,6 +43,33 @@ namespace bson_value {
 ///
 class BSONCXX_API value {
    public:
+#define BSONCXX_ENUM(name, val) value(b_##name v);
+#include <bsoncxx/enums/type.hpp>
+#undef BSONCXX_ENUM
+
+    value(const char* v);
+    value(std::string v);
+    value(stdx::string_view v);
+
+    value(int32_t v);
+    value(int64_t v);
+    value(double v);
+    value(bool v);
+    value(oid v);
+    value(decimal128 v);
+    value(std::chrono::milliseconds v);
+    value(std::nullptr_t);
+    value(bsoncxx::document::view v);
+    value(bsoncxx::array::view v);
+    value(std::vector<unsigned char> v, binary_sub_type sub_type = {});
+
+    value(const type id);
+    value(const type id, uint64_t a, uint64_t b);
+    value(const type id, stdx::string_view a, stdx::string_view b = {});
+    value(const type id, stdx::string_view a, oid b);
+    value(const type id, stdx::string_view a, bsoncxx::document::view_or_value b);
+    value(const type id, const binary_sub_type sub_id, uint32_t size, const uint8_t* data);
```

Comment:
Does this cover all types? If so, I'm ok with having these listed out as you have now, instead of trying to collapse them into one template.

Reply:
Yep, this is everything